src/workerd/jsg/setup.h

```diff
@@ -77,6 +77,9 @@ class IsolateBase {
   // Immediately cancels JavaScript execution in this isolate, causing an uncatchable exception to
   // be thrown. Safe to call across threads, without holding the lock.
 
+  void requestGcForTesting() const;
```

Comment:
I'd put this method on `jsg::Lock` rather than `IsolateBase`.

Reply:
Will do. Just curious, what is your reasoning on putting it on the `jsg::Lock` instead of `IsolateBase`

Reply:
It makes your code simpler as you don't have to do `IsolateBase::from()`.

In general `jsg::Lock` is the place for all general JavaScript runtime functionality that isn't connected to a specific object.

Reply:
> I'd put this method on jsg::Lock rather than IsolateBase.

hmm.. really? I'd prefer it not be *too* visible on an API that is more generally used. Too easy for someone to see and abuse/misuse at some point. 

I'm quite happy hiding this away on something somewhere it's less visible.

Reply:
> Too easy for someone to see and abuse/misuse at some point.

The name makes clear this is only for testing, and the implementation will not work if it's not in test mode. People "abusing" this isn't a real risk, but if it were, adding superficial road blocks doesn't seem like the right way to stop them. It just makes the code ugly.

Reply:
There is a fairly large warning on the definition in the header file. But I could see how it might be an issue, someone could use it intending that it run outside of tests and all of our testing would pass. But, the second it rolls out it would cause fatal crashes out in the wild. I do think that the name would steer people away from it, but it might be a good idea to rename to `requestGcForTestingOnly` so that it sticks out like a sore thumb in future code reviews if someone uses it.

Reply:
You make a really good point, actually... the problem with throwing an exception when called outside of tests is that it'll pass tests and fail in prod.

This might argue for changing the logic so it merely logs an error and returns (without actually running a GC) if executed in production. At least then we'll see it in Sentry.

Another possibility is to use `#ifdef KJ_DEBUG` around the whole thing and all call sites, so that the method isn't even defined in release builds. Then if the release build compiles at all we know it's not using this.

Reply:
I think that the log and return w/o executing is the cleanest way of doing this. That way we don't have to maintain an implicit contract to call `#ifdef KJ_DEBUG` all over the place. Having Sentry ping the team is actually a really good way to ensure that the function is used correctly even if someone accidentally slips it into a PR that makes it past review.

Reply:
Changed it to logging an error and returning before running the GC.